ops/model.py

```diff
@@ -1787,19 +1863,28 @@ class RelationData(Mapping[Union[Unit, Application], 'RelationDataContent']):
     :attr:`Relation.data`
     """
 
-    def __init__(self, relation: Relation, our_unit: Unit, backend: _ModelBackend):
+    def __init__(
+        self,
+        relation: Relation,
+        our_unit: Unit,
+        backend: _ModelBackend,
+        cache: _ModelCache | None = None,
+    ):
         self.relation = weakref.proxy(relation)
-        self._data: dict[Unit | Application, RelationDataContent] = {
-            our_unit: RelationDataContent(self.relation, our_unit, backend),
-            our_unit.app: RelationDataContent(self.relation, our_unit.app, backend),
+        self._data: dict[Union[Unit, Application], RelationDataContent] = {
```

Comment:
Missed in the merge:

```suggestion
        self._data: dict[Unit | Application, RelationDataContent] = {
```